src/Symfony/Component/ObjectMapper/Metadata/ReflectionObjectMapperMetadataFactory.php

```diff
@@ -0,0 +1,39 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\ObjectMapper\Metadata;
+
+use Symfony\Component\ObjectMapper\Attribute\Map;
+use Symfony\Component\ObjectMapper\Exception\MappingException;
+
+/**
+ * @internal
+ *
+ * @author Antoine Bluchet <soyuka@gmail.com>
+ */
+final class ReflectionObjectMapperMetadataFactory implements ObjectMapperMetadataFactoryInterface
+{
+    public function create(object $object, ?string $property = null, array $context = []): array
+    {
+        try {
+            $refl = new \ReflectionClass($object);
+            $mapTo = [];
+            foreach (($property ? $refl->getProperty($property) : $refl)->getAttributes(Map::class) as $mapAttribute) {
+                $map = $mapAttribute->newInstance();
+                $mapTo[] = new Mapping(source: $map->source, target: $map->target, if: $map->if, transform: $map->transform);
```

Comment:
we don't use named args exception for attributes:
```suggestion
                $mapTo[] = new Mapping($map->source, $map->target, $map->if, $map->transform);
```